Shared/Samples/Snap geometry edits with utility network rules/SnapGeometryEditsWithUtilityNetworkRulesView.swift

```diff
@@ -0,0 +1,280 @@
+    
+    /// A Boolean value indicating whether snap sources list is showing.
+    @State private var sourcesListIsPresented = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    /// The instruction text indicating the next step in the sample's workflow.
+    private var instructionText: String {
+        if state == .editing {
+            canSave
+            ? "Tap the save button to save the edits."
+            : "Tap on the map to update the feature's geometry."
+        } else {
+            model.selectedElement == nil
+            ? "Tap the map to select a feature."
+            : "Tap the edit button to update the feature."
+        }
+    }
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            MapView(map: model.map, graphicsOverlays: [model.graphicsOverlay])
+                .geometryEditor(model.geometryEditor)
+                .onSingleTapGesture { screenPoint, _ in
+                    state = .identifying(tapPoint: screenPoint)
+                }
+                .task(id: state) {
+                    // Runs the async action related to the current sample state.
+                    do {
+                        switch state {
+                        case .setup:
+                            try await model.setUp()
+                        case .identifying(let tapPoint):
+                            let identifyResults = try await mapViewProxy.identifyLayers(
+                                screenPoint: tapPoint,
+                                tolerance: 5
+                            )
+                            try await model.selectFeature(from: identifyResults)
+                        case .editing:
+                            model.startEditing()
+                            
+                            for await canUndo in model.geometryEditor.$canUndo {
+                                canSave = canUndo
+                            }
+                        case .saving:
+                            try await model.save()
+                        }
+                    } catch {
+                        self.error = error
+                    }
+                }
+                .errorAlert(presentingError: $error)
+        }
+        .overlay(alignment: .top) {
+            VStack(alignment: .trailing, spacing: 0) {
+                Text(instructionText)
+                    .multilineTextAlignment(.center)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.ultraThinMaterial, ignoresSafeAreaEdges: .horizontal)
+                
+                if let selectedElement = model.selectedElement {
+                    VStack(alignment: .leading) {
+                        LabeledContent("Asset Group:", value: selectedElement.assetGroup.name)
+                        LabeledContent("Asset Type:", value: selectedElement.assetType.name)
```

Comment:
The label is already set off from the content with a bold appearance. Using a colon as well is quite unusual.

```suggestion
                        LabeledContent("Asset Group", value: selectedElement.assetGroup.name)
                        LabeledContent("Asset Type", value: selectedElement.assetType.name)
```

Also, consider whether "Asset" is really needed. Could the labels just be "Group" and "Type"?

Reply:
I included colon because both the title and value appear the same on Mac Catalyst. See screenshot below (without colon).
![Screenshot 2025-04-01 at 9 22 30 AM](https://github.com/user-attachments/assets/7d5a17d5-dd04-4ef6-8681-9eaafd2d8dee)

Also, I would opt to keep "Asset" since it is used in the property names (`assetGroup`, `assetType`).

I honestly don't feel very strongly about either, though, so feel free to weigh in again if you still want these changes.

Reply:
That's a bummer that the weights are different on macOS. I didn't even consider that. All well.